AnsiblePlaybooks/meerkat/variables.tf

```
@@ -1,32 +1,50 @@
-variable "keypair_name" {
-    description = "The keypair to be used."
-    default  = "keyless"
+##############################################################
+### path and user info
+##############################################################
+
+variable "user"{
+    description = "username for file pathing"
+    default = "<user>"
 }
 
-variable "network_name" {
-    description = "The network to be used."
-    default  = "Internal"
+variable "playbook_path" {
+    description = "Path to playbook to be run"
+    default = "/home/<user>/cloud-ops-tools/AnsiblePlaybooks/meerkat/meerkat.yaml"
 }
 
-variable "instance_name" {
-    description = "The Instance Name to be used."
-    default  = "meerkat"
+##############################################################
+### VM details
+##############################################################
+
+variable "flavor_name" {
+    description = "List of flavor names to be used"
+    type = list
+    default  = ["l3.nano", "l3.micro"]
 }
 
 variable "image_name" {
-    description = "The image name to be used."
+    description = "List of image names to be used"
     type = list
     default  = ["ubuntu-focal-20.04-nogui"]
 }
 
-variable "flavor_name" {
-    description = "The flavor name to be used."
-    type = list
-    default  = ["l3.nano", "l3.tiny"]
+variable "keypair_name" {
+    description = "The keypair to be used"
+    default  = "keyless"
```

Can we drop this default? If nothing was specified it should fail with an error saying unset rather than it saying "the key 'keyless' doesn't exist"